File: core/domain/src/main/java/com/wansenai/dto/role/AddOrUpdateRoleDTO.java
package com.wansenai.dto.role;

import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AddOrUpdateRoleDTO {

    private Long id;

    private String roleName;

    private String type;

    private Integer priceLimit;

    private Integer status;

    private String description;

    private Long page;

    private Long pageSize;
}

File: core/domain/src/main/kotlin/com/wansenai/dto/role/AddOrUpdateRoleDTO.kt
/*
 * Copyright 2023-2025 EAIRP Team, Inc. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License"). You may not use this file except in compliance
 * with the License. A copy of the License is located at
 *
 * http://opensource.wansenai.com/apache2.0/
 *
 * or in the "license" file accompanying this file. This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES
 * OR CONDITIONS OF ANY KIND, either express or implied. See the License for the specific language governing permissions
 * and limitations under the License.
 */
package com.wansenai.dto.role

import lombok.Data
import lombok.EqualsAndHashCode

@Data
@EqualsAndHashCode(callSuper = true)
data class AddOrUpdateRoleDTO (

    var id : Long?,

    var roleName: String? ,

    var type: String?,

    var priceLimit: Int?,

    var status: Int?,

    var description: String?,

    val page: Long?,

    val pageSize: Long?
)